holopy/scattering/theory/multisphere.py

```diff
@@ -183,27 +184,15 @@ def _scsmfo_setup(self, scatterer, medium_wavevec, medium_index):
         if (centers > 1e4).any():
             raise InvalidScatterer(scatterer, "Particle separation "
                                         "too large, calculation would take forever")
-        if self.suppress_fortran_output:
-            #store default (current) stdout
-            default = os.dup(1)
-            #copy devnull into stdout
-            devnull = os.open(os.devnull,os.O_WRONLY)
-            os.dup2(devnull,1)
-
-        _, lmax, amn0, converged = scsmfo_min.amncalc(
-            1, centers[:,0],  centers[:,1],
-            # The fortran code uses oppositely directed z axis (they have laser
-            # propagation as positive, we have it negative), so we multiply the
-            # z coordinate by -1 to correct for that.
-            -1.0 * centers[:,2],  m.real, m.imag,
-            scatterer.r * medium_wavevec, self.niter, self.eps,
-            self.qeps1, self.qeps2,  self.meth, (0,0))
-
-        if self.suppress_fortran_output:
-            #restore stdout to default and close files
-            os.dup2(default,1)
-            os.close(devnull)
-            os.close(default)
+        with SuppressOutput(suppress_output=self.suppress_fortran_output):
```

Comment:
This line is pretty ugly, because it needs the output to not suppress fortran output. Does anyone actually want to see the fortran output when they are calling scattering theory? Can I just make the context manager suppress output and not have the misdirection?